### PR TITLE
Add nvidia-l4 gpu accelerator

### PR DESCRIPTION
**Pull request by eapolinario**
## Why are the changes needed?
The suffix `-vws` in the definition of the L4 gpu accelerator causes some confusion. According to google cloud [docs](https://cloud.google.com/compute/docs/gpus#l4-gpus), we can specify either `nvidia-l4` or `nvidia-l4-vws` to use L4 GPUs.

## What changes were proposed in this pull request?
Rename the existing `L4` constant to `L4_VWS` just to be safe and define a new `L4` gpu accelerator with the value set to `nvidia-l4`.

## Check all the applicable boxes 

- [ ] I updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] All commits are signed-off.

### CODE REVIEW

**Review by mhotan — APPROVED** [review]

File: flytekit/extras/accelerators.py [hunk 1]
```diff
@@ -133,7 +133,11 @@ def to_flyte_idl(self) -> tasks_pb2.GPUAccelerator:
 
 #: use this constant to specify that the task should run on an
 #: `NVIDIA L4 Tensor Core GPU <https://www.nvidia.com/en-us/data-center/l4/>`_
-L4 = GPUAccelerator("nvidia-l4-vws")
+L4 = GPUAccelerator("nvidia-l4")
```
Comment (thomasjpfan) [inline flytekit/extras/accelerators.py:136]: @mhotan Do you think changing the id can be BC breaking?

GCP is okay with both strings, but is this the same for AWS?

Comment (eapolinario) [reply flytekit/extras/accelerators.py:136]: AFAIK, we've never tested this gpu accelerator code in aws (we couldn't get those gpus at the time). Looking at the code this should work, but it requires verification.

**Comment by eapolinario** [comment]
@thomasjpfan , I also noticed that this change is going to break this `unionai` test: https://github.com/unionai/unionai/blob/main/tests/unit/actor/test_actor.py#L119

File: flytekit/extras/accelerators.py (hunk 1 above)
Comment (samhita-alla) [reply flytekit/extras/accelerators.py:136]: @eapolinario i've used `node_selector={"k8s.amazonaws.com/accelerator": "nvidia-tesla-l4"}` to get l4 to work on aws. we've l4 enabled on the demo hosted tenant.

**Review by katrogan — APPROVED** [review]

**Review by thomasjpfan — APPROVED** [review]

**Comment by thomasjpfan** [comment]
>  I also noticed that this change is going to break this unionai test:

@eapolinario I'll adjust the unit tests after this is merged.

**Pull request merged**